Lib/test/datetimetester.py

```diff
@@ -2536,6 +2536,12 @@ def test_strptime(self):
         with self.assertRaises(ValueError): strptime("-2400", "%z")
         with self.assertRaises(ValueError): strptime("-000", "%z")
 
+    def test_strptime_ampm(self):
+        for hour in range(0, 24):
+            with self.subTest(hour=hour):
+                hour_date = (1999, 3, 17, hour, 44, 55, 2, 76, 0)
+                self.assertEqual(self.theclass.strptime(_time.strftime("%I %p", hour_date), "%I %p").hour, hour)
```

Comment:
```suggestion
    def test_strptime_ampm(self):
        dt = datetime(1999, 3, 17, 0, 44, 55, 2, 76, 0)
        for hour in range(0, 24):
            with self.subTest(hour=hour):
                new_dt = dt.replace(hour=hour)
                dt_str = new_dt.strftime("%I %p")

                self.assertEqual(self.theclass.strptime(dt_str, "%I %p").hour,
                                 hour)
```
@pganssle 's suggestion